### DISCUSSION TITLE
Order of execution for migrations

### DISCUSSION THREAD

**Discussion by SpencerKaiser** [mikro-orm/mikro-orm#3119] [Q&A] [answered]
I think this might be more of an umzug question, but what's the order of operations when running migrations?

1. **Is the order determined by file name or by the class name in each migration?** 

1. **What happens if you have two with the same file number prefix?** Is it alphabetical if there are duplicate numbers?

1. **What happens if migration 5 has an earlier timestamp in the classname than migration 4?** Will 4 still run first?

**Comment by B4nan** [comment, marked as answer]
- its all about filenames, not just prefix, the whole filename
- you can't have two files with same name next to each other, so duplicities cannot coexist
- same as first point

**Reply by SpencerKaiser** [reply]
So if I have `22-do-a-thing` and `22-do-something-else`, the latter will execute first based on alphabetical sorting?

The problem I'm trying to solve for is multiple devs working on their own branches and making migrations at the same time. I want to add a new CI action to break their branch if they create migration naming conflicts. If the above is correct, I think parsing the number from the file name and looking to see if the new migration is incremented by one and throw an error if not. Any better ideas?

**Reply by B4nan** [reply]
Yeah just base it on radix order of the filenames.

**Reply by SpencerKaiser** [reply]
@B4nan so just to clarify... if I have a `01-something` with a class of `Migration20220222063928` and then `02-something-else` with a class of `Migration20220222063927` (one second earlier), `01` will execute first because of the file name regardless of whether the migration name is a lower value from a sorting perspective??

**Reply by B4nan** [reply]
i would say so, but as you said earlier, this is more of an umzug question. better try things yourself if you wanna be sure.

**Reply by SpencerKaiser** [reply]
I did try it and it worked the way I described, but I'm a little sketched out by the timestamp in the class name 😅 

I'm gonna push on forward after my test and assume it works that way. Thanks for the help, as always!!!
